code/datums/components/customizableatom.dm

```diff
@@ -0,0 +1,222 @@
+/**
+ * # Custom Atom Component
+ *
+ * When added to an atom, edible ingredients can be put into that.
+ * The sprite is updated and reagents are transfered.
+ *
+ * If the component is added to something that is processed, creating new objects (being cut, for example),
+ * the replacement type needs to also have the component. The ingredients will be copied over. Reagents are not
+ * copied over since edible components already take care of that.
+ */
+/datum/component/customizableatom
```

Comment:
if it's only for ingredients why aren't we naming it something sane like `customizable_food`?

Reply:
I will make edible ingredients a variable

Reply:
One use could be for a custom cigarette, which is not food. But this PR is for custom food, so that is not included.